src/EventHub/EventHub/Cmdlets/NetworkRuleSet/RemoveAzureEventHubVirtualNetworkrule.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.NetworkruleSet
+{
+    /// <summary>
+    /// 'New-AzureRmEventHubIpfilterRule' Cmdlet creates a new IPFilterRule
+    /// </summary>
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventHubVirtualNetworkRule", DefaultParameterSetName = VirtualNetworkRulePropertiesParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSNetworkRuleSetAttributes))]
+    public class RemoveAzureEventHubVNetRule : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRuleInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Namespace Name")]
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRuleInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Name { get; set; }        
+
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRulePropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "Subnet Resource ID")]
+        [ValidateNotNullOrEmpty]
+        public string SubnetId { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = VirtualNetworkRuleInputObjectParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "IPRule Configuration Object")]
+        [ValidateNotNullOrEmpty]
+        public PSNWRuleSetVirtualNetworkRulesAttributes VirtualNetworkRuleObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            PSNetworkRuleSetAttributes networkRuleSet = Client.GetNetworkRuleSet(ResourceGroupName, Name);
+            PSNWRuleSetVirtualNetworkRulesAttributes Toremove = new PSNWRuleSetVirtualNetworkRulesAttributes();
+
+
+            if (ParameterSetName.Equals(VirtualNetworkRuleInputObjectParameterSet))
+            {
+                Toremove = VirtualNetworkRuleObject;
+            }
+
+            if (ParameterSetName.Equals(VirtualNetworkRulePropertiesParameterSet))
+            {
+                foreach (PSNWRuleSetVirtualNetworkRulesAttributes virtualnw in networkRuleSet.VirtualNetworkRules)
+                {
+                    if (virtualnw.Subnet.Id == SubnetId)
+                        Toremove = virtualnw;
```

Comment:
@v-Ajnava same comment about adding a `break` after this assignment

Reply:
Resolved